### PR TITLE
Refactoring tab switcher

### LINKED ISSUE #283: Refactor tab switcher for mobile
Add icon-ification of tabs for narrow viewports.

### PR DESCRIPTION
**Pull request by gvn**
closes #283

### CODE REVIEW

File: app/react/components/tab-switcher/tab-switcher.jsx [hunk 1]
```diff
@@ -24,7 +24,8 @@ export default React.createClass({
           onClick={this.tabClick.bind(null, index)}
           key={index}
           hidden={this.props.children[index].props.hidden}>
-            {element.props.name}
+            <img className="hidden-sm-up" src={index === this.state.activeTab && element.props.iconActive ? element.props.iconActive : element.props.iconDefault}/>
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: This is getting a bit dense with regards to API- Can the tab itself be a component or can you add a proptype for the children elements' shape/props?

File: app/react/components/tab-switcher/tab-switcher.scss [hunk 2]
```diff
@@ -15,6 +15,11 @@
       color: $white;
       background: rgba($developer-blue, 0.7);
 
+      img {
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: I feel like this should use ems in the event we want to change the font size of tabs in the future (And really, the whole .tabs selector should probably define the font size for the rest of the children to use ems, but that's beyond the scope of this.)

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Also, use a class instead of element selector?

**Comment by alanmoo** [comment]
Close, just some nits/clarification

**Comment by gvn** [comment]
K, did those fixups...

**Comment by alanmoo** [comment]
Just wait for travis

**Pull request merged**